lib/config.js

```diff
@@ -229,13 +228,44 @@ function validateEnvConfig (prefix) {
   }
 }
 
+function deepFreeze (o) {
+  if (useTestConfig()) return o
```

Comment:
I think it would be better to put this conditional where you call this method rather than inside it.